### PR TITLE
JavaScript Group Capstone Project (development branch)

### PR DESCRIPTION
**Pull request by Thoth1111**
In this branch, we merged several feature branches that each accomplished a set of objectives from the kanban board of the project. Each branch pull request linked to one of the following issues that we accomplished:

- Setting up the project with webpack, Jest, linters, and npm installations.
- Creating a separate development branch(dev) in accordance with git flow.
- Adding and styling a header and footer for the page.
- Displaying a list of items dynamically on the home page from the mealsdb API.
- Creating a dynamic pop-up modal  for all items that shows the item details.
- Adding functions to post and fetch comments from the involvement API as well as a counter function that counts comments from the DOM and displays them in the respective modals.
- Adding functions to count the items displayed in the homepage and display the total count beside each category.
- Creating jest tests for the item and comments counters
- Styling of the page, updating the README and fixing of linter errors.

### CODE REVIEW

File: src/modules/mealsMenu.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
+import { commentBtnAction } from './modal.js';
+import { likeButtonAction, getLikesNumber } from './likes.js';
+
+const menuCategories = document.querySelector('.menuCategories');
+const menuOptions = document.querySelector('.menuOptions');
+
+// const menuListUrl = 'https://www.themealdb.com/api/json/v1/1/list.php?c=list';
```
Comment (rustam-tolipov) [inline src/modules/mealsMenu.js]: - [x] To keep your code professional. Kindly remove unused commented codes from it.

File: src/modules/likes.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+const url = 'https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/7P9rWHY7WsKk1rSNM8nF/likes';
+
+export const getLikesNumber = async () => {
```
Comment (rustam-tolipov) [inline src/modules/likes.js]: - [ ] I saw your presentation. In the video all features working as expected. However, I was not able to find like button. When I run the app in local. Make sure your like button is displaying as expected. Due to this reason I was not able to test your like button's functionality.

Your image            |  Original image
:-------------------------:|:-------------------------:
Screenshot 2023-01-03 at 18 29 21 |

**Review by rustam-tolipov — CHANGES_REQUESTED** [review]
Hello @Thoth1111, and @HanaSabih,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
_Your project is almost complete!_
_No eslint disabler_
_Perfect implementations on testing_

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Check the comments under the review._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/likes.js (hunk 2 above)
Comment (Thoth1111) [reply src/modules/likes.js]: Hello @Rustamxon7,
Thank you for the review. We will work on the changes. However, we would like a little more information about how you ran the app locally so we can identify where the problem might be. On our end, the like buttons are displaying and even so in the live [demo](https://hanasabih.github.io/the-meal/dist/), so it's a bit difficult to troubleshoot.

**Comment by Thoth1111** [comment]
Screenshot (2)

This is a screenshot of the page as it appears locally using webpack's live server.

File: src/modules/likes.js (hunk 2 above)
Comment (rustam-tolipov) [reply src/modules/likes.js]: Hi @Thoth1111, 
I run your app with `npm start`. Based on your documentation in the readme file I was not able to find the correct command to run it. I strongly suggest you add guidelines in Readme. Thank you!

Comment (Thoth1111) [reply src/modules/likes.js]: Okay, thank you. We use npm start as well. Perhaps we can request a re-review and see if the issue persists. We've removed the commented code and I'll also update the Readme now.

**Review by Shedrack-Sunday — CHANGES_REQUESTED** [review]
Hi @Thoth1111, @Rustamxon7 and @HanaSabih   👋 

# CHANGES REQUIRED ♻️

Great work on the project 👏
You've done well implementing the requirements,  However, there are a few you need to work on, to move to the next project. 💯 



Highlights
✅All linters are checked
✅ Descriptive commit messages 
  
Required Changes ♻️

- [x] You did make effort in the  Project and you really did a great job 👏 . However, it seems you reverted the changes and now the whole code from this branch has been deleted. I advise that you revert back again and make all the changes as requested by the previous reviewer.

## Previou reviewer comments:

" I saw your presentation. In the video, all features working as expected. However, I was not able to find a like button. When I run the app in locally. Make sure your like button is displaying as expected. Due to this reason, I was not able to test your like button's functionality. "

You can as well check the previous conversation and make the appropriate changes, as I can't review this branch 


_check the comment section for a detailed review_

# Optional suggestions 

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by Danieldotcomcoder — APPROVED** [review]
Hello @HanaSabih & @Thoth1111 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it 👍 👍 😄 👏 

Congratulations! 🎉 🎉 🎉 

## Highlights
- Professional Readme File ✔✔
- Linters are setup correctly ✔✔
- Clear and nice code ✔✔

comp-2
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**